--- kernel/src/sched.c
+++ kernel/src/sched.c
@@ -203,7 +203,9 @@

         // Add to the local runqueue
         uint32_t core = slot->thread.bound_core_id;
-        list_add(&slot->list_node, &g_runqueues[core].ready_queue[slot->thread.priority]);
+        if (slot->thread.priority < MAX_PRIORITY_LEVELS) {
+            list_add(&slot->list_node, &g_runqueues[core].ready_queue[slot->thread.priority]);
+        }

         return &slot->thread;
     }
